include/ccf/indexing/strategies/seqnos_by_key_in_memory.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
This is mostly `seqnos_by_key.h`, just moved here and with some of the index-construction factored out to `VisitEachEntryInMap` (so both this and `_Bucketed` implement `visit_entry`, rather than similar copies of `handle_committed_transaction`).